### ISSUE TITLE
segfault after node reboot

### ISSUE THREAD

**Issue by thatsmydoing** [aws/amazon-vpc-cni-k8s#1039]
This looks similar to https://github.com/aws/amazon-vpc-cni-k8s/issues/914 but fails in a different place.

Kubernetes: v1.15.11-eks-af3caf
AWS CNI: 1.6.0

```
starting IPAM daemon in background ... ok.
checking for IPAM connectivity ... ok.
copying CNI plugin binaries and config files ...  ok.
foregrounding IPAM daemon ... 
failed (process terminated)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1427e61]
goroutine 328 [running:
github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils.(*EC2InstanceMetadataCache).DescribeENI(0xc00012efc0, 0xc000a99020, 0x15, 0x1, 0x1, 0xc0003b2850, 0x1d91580, 0xc00062ac30, 0x0, 0x8)
    /go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils/awsutils.go:911 +0x531
github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils.(*EC2InstanceMetadataCache).freeENI(0xc00012efc0, 0xc000a99020, 0x15, 0x77359400, 0xdf8475800, 0x0, 0xc000046558)
    /go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils/awsutils.go:807 +0xed
github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils.(*EC2InstanceMetadataCache).FreeENI(0xc00012efc0, 0xc000a99020, 0x15, 0x1, 0x1)
    /go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils/awsutils.go:800 +0x57
github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd.(*IPAMContext).tryFreeENI(0xc0000e80f0)
    /go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd/ipamd.go:517 +0x11d
github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd.(*IPAMContext).updateIPPoolIfRequired(0xc0000e80f0)
    /go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd/ipamd.go:477 +0x71
github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd.(*IPAMContext).StartNodeIPPoolManager(0xc0000e80f0)
    /go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd/ipamd.go:463 +0x39
created by main._main
    /go/src/github.com/aws/amazon-vpc-cni-k8s/cmd/aws-k8s-agent/main.go:67 +0x453
stream closed
```

We've recently had problems with nodes spontaneously rebooting but I've only seen this once so far so it's not really consistent.

**Comment by mogren** [comment]
Hi @thatsmydoing, thanks for opening a ticket! 

I see that you use v1.6.0, are you absolutely sure it is not the same as issue #914? This is the line from your log:

https://github.com/aws/amazon-vpc-cni-k8s/blob/20c3a6519cf96c00a3d3dd6ae6e92c24fc4829b6/pkg/awsutils/awsutils.go#L911

It seems to assume that `result.NetworkInterfaces[0].Attachment` will never be `nil`, which not  the case. Could you try updating to v1.6.2 and see if that fixes the issue?

**Comment by thatsmydoing** [comment]
Ah, sorry. The linked issue mentioned the fix being in 1.6, but I didn't realize that it was merged in 1.6.1 instead. I'll just close the issue and reopen it if I see it again after updating.
